docs/guides/validation-user-guide.md

```diff
@@ -4,4 +4,461 @@ headline: Validation User Guide
 bodyclass: docs
 layout: docs
 ---
-<p class="coming-soon">Coming soon...</p>
+
+Protobuf is a great tool for structuring domain models. Protobuf definitions help us to standardize
+creating [Value Objects](https://martinfowler.com/bliki/ValueObject.html).
+Spine enhances Protobuf with a validation library.
+
+This guide will walk you though the API of Spine validation library. All of the validation features
+described here are currently supported in the Java environment. Many are supported in Dart as well.
+For more info, see the description of individual constraints. 
+
```

Comment:
We need a short overview of the goals of the Validation library. Why do people would want to use it?

Then we need to describe how it's done in general. Then go to options.